text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,49 @@
+# Implement a Bill of Materials Across Paketo
+
+## Summary
+
+A bill of materials (BOM) is a set of metadata that provides an account of the
+dependencies and tools that are both present in an application image, as well
+those used in the construction of that image. The availability of this data is
+a key value proposition of Cloud Native Buildpacks.
+
+As an implementation of Cloud Native Buildpacks, the Paketo project should
+support adding bill of materials metadata across the different components that
+go into an app image.
+
+## Motivation
+
+A BOM can include an array of different metadata about application dependencies
+and buildpack dependencies such as SHAs, URIs, versions, license information,
```

Comment:
If the dependency is added to a buildpack-provided layer (rather than the app dir) java buildpacks currently include the name of the associated layer in the BOM entry. Maybe we should make this a standard practice across paketo buildpacks? Although the lifecycle might [help us out with this](https://github.com/buildpacks/rfcs/pull/151/files#diff-d31996aeb587aab806ae658bfb0199878b47d25545090827fa6b81723ca2d033R143) in the future.

Reply:
What benefit does this provide to an end user inspecting the BOM? Is there a consumer of this field that I am unaware of?

Reply:
I'm not 100% certain, but if you know the layer in with the item exists, you can tell what installed it. If you were trying to track down an issue with something that was installed, such as a dependency that wasn't the expected version that information would be helpful to know.

Reply:
Given that the upstream RFC to include BOM metadata in the layer was accepted, should we hold off on a requirement to specify the layer in the BOM entry itself?